x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/SyntheticSourceIndexSettingsProvider.java

```diff
@@ -163,4 +169,33 @@ private IndexMetadata buildIndexMetadataForMapperService(
         tmpIndexMetadata.settings(finalResolvedSettings);
         return tmpIndexMetadata.build();
     }
+
+    /**
+     * The GA-ed use cases in which synthetic source usage is allowed with gold or platinum license.
+     */
+    boolean isLegacyLicensedUsageOfSyntheticSourceAllowed(IndexMode templateIndexMode, String indexName, String dataStreamName) {
+        if (templateIndexMode == IndexMode.TIME_SERIES) {
+            return true;
+        }
+
+        if (dataStreamName != null && dataStreamName.startsWith("profiling-metrics")) {
+            return true;
+        } else if (dataStreamName != null && dataStreamName.startsWith("profiling-events")) {
+            return true;
+        } else if (indexName.startsWith(".profiling-sq-executables")) {
+            return true;
+        } else if (indexName.startsWith(".profiling-sq-leafframes")) {
+            return true;
+        } else if (indexName.startsWith(".profiling-stacktraces")) {
+            return true;
+        }
+
+        // To allow the following patterns: metrics-apm.transaction.*, metrics-apm.service_transaction.*, metrics-apm.service_summary.*,
```

Comment:
nit: I would maybe just collect all these exceptions into one or more list of strings

Reply:
I also think that a broader prefix is more preferable.